packages/router-core/src/ssr/ssr-client.ts

```diff
@@ -1,113 +1,77 @@
 import invariant from 'tiny-invariant'
-import { isPlainObject } from '../utils'
-import { tsrSerializer } from '../serializer'
-import type { DeferredPromiseState } from '../defer'
-import type { MakeRouteMatch } from '../Matches'
-import type { AnyRouter, ControllablePromise } from '../router'
+import type { AnyRouteMatch, MakeRouteMatch } from '../Matches'
+import type { AnyRouter } from '../router'
 import type { Manifest } from '../manifest'
 import type { RouteContextOptions } from '../route'
+import type { GLOBAL_TSR } from './ssr-server'
 
 declare global {
   interface Window {
-    __TSR_SSR__?: TsrSsrGlobal
+    [GLOBAL_TSR]?: TsrSsrGlobal
   }
 }
 
-export interface TsrSsrGlobal {
-  matches: Array<SsrMatch>
-  streamedValues: Record<
-    string,
-    {
-      value: any
-      parsed: any
+declare module '../router' {
+  interface RouterEvents {
+    onStreamedValue: {
+      type: 'onStreamedValue'
+      key: string
     }
-  >
-  cleanScripts: () => void
-  dehydrated?: any
-  initMatch: (match: SsrMatch) => void
-  resolvePromise: (opts: {
-    matchId: string
-    id: number
-    promiseState: DeferredPromiseState<any>
-  }) => void
-  injectChunk: (opts: { matchId: string; id: number; chunk: string }) => void
-  closeStream: (opts: { matchId: string; id: number }) => void
-}
-
-export interface SsrMatch {
-  id: string
-  __beforeLoadContext: string
-  loaderData?: string
-  error?: string
-  extracted?: Array<ClientExtractedEntry>
-  updatedAt: MakeRouteMatch['updatedAt']
-  status: MakeRouteMatch['status']
-  ssr?: boolean | 'data-only'
-}
-
-export type ClientExtractedEntry =
-  | ClientExtractedStream
-  | ClientExtractedPromise
-
-export interface ClientExtractedPromise extends ClientExtractedBaseEntry {
-  type: 'promise'
-  value?: ControllablePromise<any>
+  }
 }
-
-export interface ClientExtractedStream extends ClientExtractedBaseEntry {
-  type: 'stream'
-  value?: ReadableStream & { controller?: ReadableStreamDefaultController }
+export interface TsrSsrGlobal {
+  r?: DehydratedRouter
```

Comment:
While it makes sense for the `matches` to have keys be shortened (i.e. `i`, `b`, `l`, etc.), I think these single callers may benefit from just being left with having descriptive names to represent their purpose (i.e. `router/dehydratedRouter`, `close`, `whatever_v_is`).